esx_service/vmci/vmci_client.c

```diff
@@ -360,7 +360,7 @@ Vmci_GetReply(int port, const char* json_request, const char* be_name,
       return CONN_FAILURE;
    }
 
-   req.mlen = strnlen(json_request, MAXBUF) + 1;
+   req.mlen = strnlen(json_request, MAXBUF-1) + 1;
```

Comment:
@govint - thanks for pointing to this. While this +- issue could not have been a reason for test failing (see below) , the buffer size control needs fixing, albeit in a separate PR.

Here is the summary : 

> the original code was creating a string which was MAXBUF + 1,

Actually it was not - the string was created in GO with whatever size, and the code tried to  make sure that strings > 1M were  truncated before being passed to VMCI but the actual truncation is missing (indeed a bug). 
However, this was never hit, because  vmdk_opsd.py MAX_JSON_SIZE is set to 4K, and for    vmci message > 4K the vmci_server.c receive() code will reject the message and drop the connection with "message too large" , without even trying to receive the message body !

The bottom line:
* Please remove this line from the PR  and simply enable the test. LGTM from me on that
* I also suggest **a separate PR** that cleans up the inconsistency. I suggest making MAXBUF consistent with MAX_JSON_SIZE, put comments in the code with cross-reference, and then limit strings to MAXBUF-1 plus NULL termination - 

In this new PR this specific text would change to
```
   req.mlen = strnlen(json_request, MAXBUF - 1) + 1;
   req.msg = json_request;       // sizeof string, with null temintation
   req.msg[req.mlen] = NULL;  // just in case we hit the MAXBUF limit
```
```